docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,128 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+User Agent Filter module allows you to allow or deny traffic based on the user-agent request header. You can define allow and deny regular expression rules that apply to the incoming user-agent request header.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### SSH
+
+```bash
+ssh -R 443:localhost:80 connect.ngrok-agent.com http \
+ --ua-filter-allow="(foo/(\d)+.(\d)+)" \
+ --ua-filter-deny="(bar/(\d)+.(\d)+)" \
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+ example:
+   proto: https
+   addr: 80
+   ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+   ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
+
+```go
+import (
+	"context"
+	"net"
+
+	"golang.ngrok.com/ngrok"
+	"golang.ngrok.com/ngrok/config"
+)
+
+func listenUserAgentFIlter(ctx context.Context) net.Listener {
+	listener, _ := ngrok.Listen(ctx,
+		config.HTTPEndpoint(
+			config.WithUserAgentFilter( []string{"(foo)/(\d)+.(\d)+"}, []string{"bar/(\d)+.(\d)+"}),
+		),
+		ngrok.WithAuthtokenFromEnv(),
+	)
+	return listener
+}
+```
+
+## Behavior
+
+### Rule Evaluation
+
+On request, the User Agent Filter module will check the incoming `user-agent` header value against the list of defined allow and deny regular expression rules.
+
+Requests with no `user-agent` header specified will still be matched against the list of defined rules. If you wish to reject these you can add a deny rule for empty strings: `^$`
+
+Requests with multiple `user-agent` headers will be matched as a single value.
+
+### Match Behavior
+
+Requests are allowed if they match an allow rule OR they match no deny rules. Requests matching deny rules are rejected with a 403 Forbidden error.
+
+You can override `deny` rules by defining an `allow` rule that is more specific, for instance blocking all Google bots except Google Images:
+
+## Reference
+
+### Configuration
+
+###### **Agent Configuration**
+
+| Parameter                   | Description                                                   |
+| --------------------------- | ------------------------------------------------------------- |
+| **User Agent Filter Allow** | A set of regular expressions used to match allow User-Agents. |
+| **User Agent Filter Deny**  | A set of regular expressions used to match deny User-Agents.  |
```

Comment:
nit:
>used to match deny User-Agents
reads a bit weird to me, maybe something like 
"A set of regular expressions used to match User-Agents that will be denied"